### PR TITLE
Added check and method for neovim api info

### PR DESCRIPTION
**Pull request by bert88sta**
If this needs any changes, let me know

### CODE REVIEW

File: browser/src/NeovimInstance.ts [hunk 1]
```diff
@@ -126,6 +126,14 @@ export class NeovimInstance extends EventEmitter implements INeovimInstance {
                             this.emit("event", eventName, eventContext)
                         } else if (pluginMethod === "window_display_update") {
                             this.emit("window-display-update", args[0][0], args[0][1])
+                        } else if (pluginMethod === "api_info") {
+                            const apiVersion = args[0][0]
+                            if (apiVersion.minor < 1) {
```
Comment (kopischke) [inline browser/src/NeovimInstance.ts]: If you target API version 0.1.6 or above, shouldn't this be `< 2`?

Comment (kopischke) [inline browser/src/NeovimInstance.ts]: … also, thinking about it, shouldn't this be able to handle the case the Neovim API hits a major version > 0? I.e.
```javascript
if (apiVersion.major < 1 && apiVersion.minor < 2 && apiVersion.patch < 6) {
  // handle incompatible version
}

Comment (justinmk) [inline browser/src/NeovimInstance.ts]: Only the API level should be checked (that's the entire purpose of having an API level). The nvim major/minor versions are irrelevant regarding API compatibility.

Comment (justinmk) [inline browser/src/NeovimInstance.ts]: Also these fields didn't exist in api_info result before 0.1.6.

Comment (kopischke) [inline browser/src/NeovimInstance.ts]: @justinmk sorry, I didn't follow the whole API level versioning discussion closely : the version numbers are not semantic? Then my comments are academic, if course: true as to pure code logic but irrelevant to the issue at hand ;).

Comment (bert88sta) [inline browser/src/NeovimInstance.ts]: I'll go in and fix this later, it was pretty late when I wrote this

**Comment by justinmk** [comment]
@kopischke the version numbers are semantic. The API level is more correct for comparing API capabilities. API level might not change between Nvim releases.

File: browser/src/NeovimInstance.ts [hunk 2]
```diff
@@ -126,6 +126,12 @@ export class NeovimInstance extends EventEmitter implements INeovimInstance {
                             this.emit("event", eventName, eventContext)
                         } else if (pluginMethod === "window_display_update") {
                             this.emit("window-display-update", args[0][0], args[0][1])
+                        } else if (pluginMethod === "api_info") {
+                            const apiVersion = args[0][0]
```
Comment (justinmk) [inline browser/src/NeovimInstance.ts]: The fields won't exist at all in older versions

File: vim/core/oni-core-interop/plugin/init.vim [hunk 3]
```diff
@@ -113,3 +113,13 @@ function OniUpdateWindowDisplayMap()
...
     call OniNotify(["window_display_update", context, mapping])
 endfunction
+
+function OniApiInfo()
+    if (has_key(api_info(),'version'))
+        call OniNotify(["api_info",api_info()['version']])
+    else
+        call OniNotify(["api_info",{api_level:0}])
+    endif
+
```
Comment (extr0py) [inline vim/core/oni-core-interop/plugin/init.vim]: Could you please remove these extra newlines? No linting set up for us in VimL :)

**Comment by extr0py** [comment]
Looks good except for some minor whitespace issues in `init.vim`. I'll pull this in once that's addressed. Thanks for the PR, @bert88sta !

**Comment by extr0py** [comment]
Thanks @bert88sta for the fix, and @kopischke & @justinmk for the feedback!

**Pull request merged**